lib/kv-redis.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var assert = require('assert');
+var Connector = require('loopback-connector').Connector;
+var debug = require('debug')('loopback:connector:kv-redis');
+var Redis = require('ioredis');
+var Command = Redis.Command;
+var util = require('util');
+
+exports.initialize = function initializeDataSource(dataSource, callback) {
+  var settings = dataSource.settings;
+
+  // TODO handle settings.lazyConnect - use ioredis option "lazyConnect"
+  dataSource.connector = new RedisKeyValueConnector(settings);
+
+  if (!callback) return;
+
+  dataSource.connector._client
+    .once('connect', function() { callback(); })
```

Comment:
If the `connect` event includes a payload it would be passed as the `error` argument to `callback` if it was passed in like that.